features/step_definitions/myStepDefinitions.js

```diff
@@ -179,9 +179,25 @@ module.exports = function () {
     }, parseInt(expected, 10))
   })
 
-  this.When(/^I type in "([^"]*)"$/, function (input, callback) {
+  this.When(/^I type in "([^"]*)"$/, function (input) {
     this.type(input)
-    callback()
+    return eventually(() => {
+      return this.getQuery()
+    }, input)
```

Comment:
Since type isn't a promise, we create an eventually that waits until the input in the field is the same input you wanted.